hera_pspec/utils.py

```diff
@@ -525,3 +569,284 @@ def flatten(nested_list):
     """
     return [item for sublist in nested_list for item in sublist]
 
+
+def config_pspec_blpairs(uv_templates, pol_pairs, group_pairs, exclude_auto_bls=True, 
+                         exclude_permutations=True, bl_len_range=(0, 1e10), 
+                         bl_deg_range=(0, 180), xants=None, verbose=True):
+    """
+    Given a list of miriad file templates and selections for
+    polarization and group labels, construct a master list of
+    blpair-group-pol pairs using utils.construct_reds().
+
+    A group is a fieldname in the visibility files that denotes the
+    "type" of dataset. For example, the group field in the following files
+        zen.even.LST.1.01.xx.HH.uv
+        zen.odd.LST.1.01.xx.HH.uv
+    are the "even" and "odd" field, and specifies the two time binning groups.
+    To form cross spectra between these two files, one would feed a group_pair
+    of: group_pairs = [('even', 'odd'), ...].
+
+    A baseline-pair is formed by self-matching unique-files in the
```

Comment:
So it will do the first, but not the second based on how you had it: i.e. it doesn't permute the keys for you. if you wanted to do both, you should feed:
`group_pairs = [('even', 'odd'), ('odd', 'even')]`